mava/systems/tf/mappo/system.py

```diff
@@ -50,28 +53,33 @@ def __init__(
             DecentralisedValueActorCritic
         ] = DecentralisedValueActorCritic,
         trainer_fn: Type[training.MAPPOTrainer] = training.MAPPOTrainer,
-        executor_fn: Type[core.Executor] = execution.MAPPOFeedForwardExecutor,
+        executor_fn: Type[
+            Union[execution.MAPPOFeedForwardExecutor, execution.MAPPORecurrentExecutor]
+        ] = execution.MAPPOFeedForwardExecutor,
         num_executors: int = 1,
-        num_caches: int = 0,
+        trainer_networks: Union[
+            Dict[str, List], enums.Trainer
+        ] = enums.Trainer.single_trainer,
+        network_sampling_setup: Union[
+            List, enums.NetworkSampler
+        ] = enums.NetworkSampler.fixed_agent_networks,
         environment_spec: mava_specs.MAEnvironmentSpec = None,
         shared_weights: bool = True,
-        agent_net_keys: Dict[str, str] = {},
         executor_variable_update_period: int = 100,
         policy_optimizer: Union[
             snt.Optimizer, Dict[str, snt.Optimizer]
         ] = snt.optimizers.Adam(learning_rate=5e-4),
         critic_optimizer: snt.Optimizer = snt.optimizers.Adam(learning_rate=1e-5),
-        discount: float = 0.99,
-        lambda_gae: float = 0.99,
+        discount: float = 0.999,
+        lambda_gae: float = 1.0,
         clipping_epsilon: float = 0.2,
         entropy_cost: float = 0.01,
         baseline_cost: float = 0.5,
         max_gradient_norm: Optional[float] = None,
-        max_queue_size: int = 100000,
-        batch_size: int = 256,
-        sequence_length: int = 10,
-        sequence_period: int = 5,
-        max_executor_steps: int = None,
+        max_queue_size: int = 1000,
+        batch_size: int = 32,
+        sequence_length: int = 20,
+        sequence_period: int = 10,
```

Comment:
Maybe we can set this to 19 / sequence_length-1 - https://github.com/deepmind/acme/blob/93a6a15b8c7b1d6a9c8099b9f2e9c693ab11a41e/acme/agents/jax/ppo/builder.py#L110 .

Reply:
Interesting :slightly_smiling_face: I will have to look into this. Thanks for pointing this out.

Reply:
Done.